src/storer.ts

```diff
@@ -166,12 +132,53 @@ export class Storer {
             }
             this.blobService.createBlockBlobFromText(this.container, blobName, compressedBlobContent, (err) => {
                 if(callback) {
-                    callback(err);
+                    return callback(err);
                 }
+                callback();
             });
-
         });
 
         this.blob = undefined;
     }
+
+    public flushDatabase(callback ? : (err? : Error) => void) : void {
+        const websiteInserts = [];
+        const containsInserts = [];
+        this.websites.forEach(website => {
```

Comment:
[http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html](http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html)
Don't mess with my forEach loops :heart: